cogito/protocol.go

```diff
@@ -21,6 +21,20 @@ type CheckInput struct {
 	Env     Environment
 }
 
+// GetInput is the JSON object passed to the stdin of the "in" executable plus
+// build metadata (environment variables).
+//
+// See https://concourse-ci.org/implementing-resource-types.html#resource-in
+//
+type GetInput struct {
+	Source  Source  `json:"source"`
+	Version Version `json:"version"`
+	// Cogito does not support get params; a resource supporting them would have the
+	// following line uncommented:
+	// Params  GetParams `json:"params"`
```

Comment:
I left this commented out field on purpose, to have `GetInput` as a blueprint for writing a resource. The same "blueprint-thinking" can be seen below, for `GetParams`.